packages/amplify-cli-core/src/customPoliciesUtils.ts

```diff
@@ -0,0 +1,77 @@
+import { Fn, IAM } from "cloudform-types";
+import * as iam from '@aws-cdk/aws-iam';
+import { JSONUtilities, pathManager } from ".";
+import * as fs from 'fs-extra';
+
+export type CustomIAMPolicies = CustomIAMPolicy[];
+
+export type CustomIAMPolicy = {
+    Action: string[];
+    Effect: string;
+    Resource: string[];
+}
+
+
+export const CustomIAMPoliciesSchema = {
+    type : "array",
+    minItems: 1,
+    items: {type: "object"},
```

Comment:
items should be typed as CustomIAMPolicySchema

Reply:
I tried the CustomIAMPolicy and CustomIAMPolicySchema, I got an error saying this is not a valid type.

Reply:
you'll need to nest them together. Ie the CustomIAMPolicySchema should fully define the expected structure of the JSON

Reply:
Changed